lib/attr_extras/attr_initialize.rb

```diff
@@ -12,22 +12,13 @@ def apply
     names = @names
     block = @block
     validate_arity = method(:validate_arity)
-    set_ivar_from_hash = method(:set_ivar_from_hash)
+    set_ivar_from_args = method(:set_ivar_from_args)
 
     klass.send(:define_method, :initialize) do |*values|
       validate_arity.call(values.length, self.class)
 
       names.zip(values).each do |name_or_names, value|
```

Comment:
Hey @bartoszkopinski, lost track of this and just stumbled across it again. So sorry.

Would you be up for modifying this pull request so it has the test I proposed and passes it, and also a test for the other problem I described, if the first test doesn't cover that?

We use this code a lot and it has decent adoption, so I want to be very confident that we're not breaking things if we add this feature.

If you don't want to put in the additional effort after this time I understand :/ It's definitely on my radar, and if we don't get it in now, I might attempt it myself on some future hack night.